contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -600,6 +743,9 @@ abstract contract PoSValidatorManager is
 
             uint256 reward = _calculateDelegationReward(delegator);
             $._redeemableDelegatorRewards[delegationID] = reward;
+            if (rewardRecipient != address(0)) {
```

Comment:
Is there a reason both this block and the below one can't be repeated just once outside of this `if/else`?

Reply:
I added to the `_calculateDelegationReward` function instead. Otherwise, we would have to do the same check for the status, which is a little more fragile IMO (if the condition changes in the if/else, the author has to remember to update the next block). 

In any case, I reduced the repetition.